internal/ast/externalmoduleindicator.go

```diff
@@ -0,0 +1,137 @@
+package ast
+
+import (
+	"github.com/microsoft/typescript-go/internal/core"
+	"github.com/microsoft/typescript-go/internal/tspath"
+)
+
+type ExternalModuleIndicatorOptions struct {
+	jsx   bool
+	force bool
+}
```

Comment:
Shifting this code around now means that the entire cache key for the external module indicator is just these two booleans! So much more reuse.